### PR TITLE
Add tool to conver disk image into Docker image

### LINKED ISSUE #11575: Develop tool to convert a virtual disk image to a docker image
**Problem**
 
Develop tool to convert a virtual disk image to a docker image
 
**Solution**
 
Use a tool like qemu-nbd (or equivalent) to mount a virtual disk, extract
 the filesystem, and convert that into a single large docker image.  Bonus
 points for allowing the user to specify a base image to start with
 that's "close" to the original OS image and then ensure the single delta
 image contains only the changes and not the entire disk content.

### PR DESCRIPTION
**Pull request by maximkulkin**
dockerize-image tool takes a virtual disk image file
and creates a Docker image based on it. You can
specify a base Docker image to make this tool create
an image that will contain only filesystem diff
instead of full filesystem.
See tools usage for details.

Fixes #11575

### CODE REVIEW

**Comment by duglin** [comment]
+1 !!
Any chance of getting some tests for this?

**Comment by maximkulkin** [comment]
Are there any examples of such tests?

**Comment by duglin** [comment]
hmm, can't think of any off-hand, but I would think you could simply create a new one in integration-cli and have it run your tool and test the output.

**Comment by rrarunan** [comment]
@maximkulkin Can you please add a `closes` or `fixes` keyword so its easy to track the fix? Thanks!

**Comment by crosbymichael** [comment]
This is cool.  Nice work.

LGTM

File: contrib/dockerize-image [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
```
Comment (tianon) [inline contrib/dockerize-image]: Can we add `set -e` here to make sure that if something fails, the script doesn't continue barreling forward? :smile:

**Comment by tianon** [comment]
This is neat!  I think the `contrib/dockerize-image` name is a little misleading though, given that "images" are a thing in Docker parlance too.  Maybe something like `contrib/dockerize-disk` ?

**Comment by thaJeztah** [comment]
Very nice! Thanks for your contribution!

**Comment by maximkulkin** [comment]
I think it still has issues (e.g. I'm not happy of using the same network block device - /dev/nbd0 - for mounting disk, but I haven't found a more elegant solution for that except for allowing user to specify which device to use which is not very elegant either), but I think it's good for the first iteration.
The other thing: it requires to be run under root (or permissions set up to allow binding NBDs and doing mounts). If you have thoughts on it, I would be happy to discuss it.

**Comment by aluzzardi** [comment]
Neat! I think this is fine given the concerns you have considering it's a first iteration.

LGTM

@tianon As resident bash-magician, any comments on the code itself?

**Comment by icecrime** [comment]
Nice! I think @tianon's comments were already taken into account, so let's merge this :tada:

**Comment by smartberry1122** [comment]
Nice

**Pull request merged**